pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -905,9 +909,11 @@ public void connectionOpened(final ClientCnx cnx) {
                     // consumer was subscribed and connected but we got some error, keep trying
                     reconnectLater(e.getCause());
```

Comment:
hi, i notice the `Connection` impl will have a chance to do reconnect logic when call `connectionOpened` it seems unnecessary now. what reconnection logic do we expect ? always handle by ConnectionHandler or by `Connection` impl themself ?

Reply:
Good catch, I added `duringConnect.set(false)` in `reconnectLater` to ensure the `duringConnect` is false when the `reconnectLater` method is called outside by the `Connection` implementations.

Ideally, it's better to disallow the implementation class to call `reconnectLater` but it's another topic.